src/lib/transaction_snark/test/zkapps_examples/initialize_state/initialize_state.ml

```diff
@@ -0,0 +1,271 @@
+          | fee_payer ->
+              fee_payer
+        in
+        let other_parties =
+          Parties.Call_forest.map other_parties ~f:(function
+            | ({ body = { public_key; use_full_commitment; _ }
+               ; authorization = Signature _
+               } as party :
+                Party.t)
+              when Public_key.Compressed.equal public_key pk_compressed ->
+                let commitment =
+                  if use_full_commitment then full_commitment
+                  else transaction_commitment
+                in
+                { party with
+                  authorization =
+                    Signature
+                      (Schnorr.Chunked.sign sk
+                         (Random_oracle.Input.Chunked.field commitment))
+                }
+            | party ->
+                party)
+        in
+        { fee_payer; other_parties; memo }
+      in
+      let parties : Parties.t =
+        sign_all { fee_payer; other_parties = ps; memo }
+      in
+      Ledger.with_ledger ~depth:ledger_depth ~f:(fun ledger ->
+          let account =
+            Account.create account_id
+              Currency.Balance.(
+                Option.value_exn (add_amount zero (Currency.Amount.of_int 500)))
+          in
+          let _, _ =
+            Ledger.get_or_create_account ledger account_id account
+            |> Or_error.ok_exn
+          in
+          let target_ledger = apply_parties ledger [ parties ] in
+          Sparse_ledger.find_index_exn target_ledger account_id
+          |> Sparse_ledger.get_exn target_ledger)
+
+    let%test_unit "Initialize" =
+      let account =
+        test_parties [ Deploy_party.party; Initialize_party.party ]
+      in
+      let zkapp_state = (Option.value_exn account.zkapp).app_state in
+      Pickles_types.Vector.iter
+        ~f:(fun x -> assert (Snark_params.Tick.Field.(equal zero) x))
+        zkapp_state
+
+    let%test_unit "Initialize and update" =
+      let account =
+        test_parties
+          [ Deploy_party.party
+          ; Initialize_party.party
+          ; Update_state_party.party
+          ]
+      in
+      let zkapp_state = (Option.value_exn account.zkapp).app_state in
+      Pickles_types.Vector.iter
+        ~f:(fun x -> assert (Snark_params.Tick.Field.(equal one) x))
+        zkapp_state
+
+    let%test_unit "Initialize and multiple update" =
+      let account =
+        test_parties
+          [ Deploy_party.party
+          ; Initialize_party.party
+          ; Update_state_party.party
+          ; Update_state_party.party
+          ]
+      in
+      let zkapp_state = (Option.value_exn account.zkapp).app_state in
+      Pickles_types.Vector.iter
+        ~f:(fun x -> assert (Snark_params.Tick.Field.(equal one) x))
+        zkapp_state
+
+    let%test_unit "Update without initialize fails" =
+      let account =
+        test_parties [ Deploy_party.party; Update_state_party.party ]
+      in
+      assert (Option.is_none account.zkapp)
```

Comment:
add a case where the zkapp account is new? in which case the account shouldn't be created at all